File: solutions/JavaScript/median_sorted_arrays_two_pointer.js
function findMedianSortedArrays(nums1, nums2) {
    const merged = [];
    let i = 0, j = 0;

    while (i < nums1.length && j < nums2.length) {
        if (nums1[i] <= nums2[j]) {
            merged.push(nums1[i++]);
        } else {
            merged.push(nums2[j++]);
        }
    }

    while (i < nums1.length) merged.push(nums1[i++]);
    while (j < nums2.length) merged.push(nums2[j++]);

    const n = merged.length;
    return n % 2 === 1 ? merged[Math.floor(n / 2)] :
        (merged[n / 2 - 1] + merged[n / 2]) / 2;
}

File: solutions/Python/median_sorted_arrays_two_pointer.py
from typing import List

def findMedianSortedArrays(nums1: List[int], nums2: List[int]) -> float:
    """Two-pointer merge approach: O(m+n) time, O(1) space"""
    merged = []
    i, j = 0, 0

    while i < len(nums1) and j < len(nums2):
        if nums1[i] <= nums2[j]:
            merged.append(nums1[i])
            i += 1
        else:
            merged.append(nums2[j])
            j += 1

    merged.extend(nums1[i:])
    merged.extend(nums2[j:])

    n = len(merged)
    if n % 2 == 1:
        return float(merged[n // 2])
    else:
        return (merged[n // 2 - 1] + merged[n // 2]) / 2.0
